### PR TITLE
Feature/df 1204 validate avam occupation code

### PR DESCRIPTION
**Pull request by fabiangotzen**

### CODE REVIEW

File: web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java [hunk 1]
```diff
@@ -58,10 +63,25 @@
...
+    private JobCenterService jobCenterService;
+
+
     @Before
     public void setUp() {
         this.jobAdvertisementRepository.deleteAll();
         this.jobAdvertisementElasticsearchRepository.deleteAll();
+        when(locationService.isLocationValid(ArgumentMatchers.any())).thenReturn(true);
+        when(locationService.enrichCodes(ArgumentMatchers.any())).then(returnsFirstArg());
+        when(professionService.isKnownAvamCode(ArgumentMatchers.any())).thenReturn(Boolean.TRUE);
```
Comment (emina-str) [inline web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java]: why not just primitive value here?

Comment (fabiangotzen) [reply web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java]: overlooked it, primitive value suits better. Changed it back now.

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java [hunk 2]
```diff
@@ -88,926 +89,930 @@
...
+		return JobAdvertisementDto.toDto(jobAdvertisement);
+	}
+
+	public JobAdvertisementDto findByStellennummerEgov(String stellennummerEgov) {
+		Optional<JobAdvertisement> jobAdvertisement = jobAdvertisementRepository.findByStellennummerEgov(stellennummerEgov);
+		return jobAdvertisement.map(JobAdvertisementDto::toDto).orElse(null);
+	}
+
+	public JobAdvertisementDto getByStellennummerEgovOrAvam(String stellennummerEgov, String stellennummerAvam) {
+		final String externalId = hasText(stellennummerEgov) ?
```
Comment (birom) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: IMHO, extracting this to separate method does not make the code simpler. I woud either leave it as it was earlier or do something like this:

```
private Occupation enrichOccupationWithProfessionCodes(Occupation occupation) {
		Condition.notNull(occupation, "Occupation can't be null");

		Function<Profession, Occupation> mapProfessionToOccupation = profession -> new Occupation.Builder()
				.setAvamOccupationCode(occupation.getAvamOccupationCode())
				.setSbn3Code(profession.getSbn3Code())
				.setSbn5Code(profession.getSbn5Code())
				.setBfsCode(profession.getBfsCode())
				.setLabel(profession.getLabel())
				.setWorkExperience(occupation.getWorkExperience())
				.setEducationCode(occupation.getEducationCode())
				.setQualification(occupation.getQualificationCode())
				.build();

		return professionService.findByAvamCode(occupation.getAvamOccupationCode())
				.map(mapProfessionToOccupation)
				.orElse(occupation);
	}
```

Comment (fabiangotzen) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: Changed to:

```
	private Occupation enrichOccupationWithProfessionCodes(Occupation occupation) {
		Condition.notNull(occupation, "Occupation can't be null");
		return professionService.findByAvamCode(occupation.getAvamOccupationCode())
				.map(profession -> new Occupation.Builder()
						.setAvamOccupationCode(occupation.getAvamOccupationCode())
						.setSbn3Code(profession.getSbn3Code())
						.setSbn5Code(profession.getSbn5Code())
						.setBfsCode(profession.getBfsCode())
						.setLabel(profession.getLabel())
						.setWorkExperience(occupation.getWorkExperience())
						.setEducationCode(occupation.getEducationCode())
						.setQualification(occupation.getQualificationCode())
						.build())
				.orElse(occupation);
	}

**Pull request merged**